frontend/src/metabase/actions/components/ActionViz/Action.unit.spec.tsx

```diff
@@ -39,15 +39,30 @@ const ACTION = createMockQueryAction({
   parameters: [
     createMockActionParameter({
       id: "parameter_1",
-      type: "type/Text",
+      name: "Parameter 1",
+      type: "string/=",
       target: ["variable", ["template-tag", "1"]],
     }),
     createMockActionParameter({
       id: "parameter_2",
-      type: "type/Integer",
+      name: "Parameter 2",
+      type: "number/=",
       target: ["variable", ["template-tag", "2"]],
     }),
   ],
+  visualization_settings: {
+    fields: {
+      // TODO: mock these properly
+      parameter_1: {
+        fieldType: "string",
+        inputType: "string",
```

Comment:
@iethree to fix some tests, I have to add `visualization_settings.fields` to the parameters. I haven't added all the data in the field yet, because I feel like there could be a better way to mock these. That's why it's not type checking. What do you suggest?

Also here as well:
https://github.com/metabase/metabase/pull/30356/files#diff-fe52934e6c2f8d9dbc22bfc529903b0d3ec274aa699b34eb0ea66edbaa4acb75R38

Reply:
🤔  yeah this is a tough problem: e.g. you might want to turn the mock into some sort of factory that generates `visualizations_settings.fields` based on the parameters - which is what the application code does. Once mocks get sufficiently complex, you almost need to test them too, which I think we should avoid by keeping mocks as dumb as possible.

I think you should just use `createMockFieldSettings` here:

```ts
  visualization_settings: {
    fields: {
      parameter_1: createMockFieldSettings({
        id: "parameter_1",
        placeholder: "Parameter 1 placeholder",
      }),
      parameter_2: createMockFieldSettings({
        id: "parameter_1",
        placeholder: "Parameter 2 placeholder",
      }),
    },
  },
```
This will make sure the objects have all the right properties with default data, and is therefore typed correctly

Reply:
Alright, I've done just that: [Fix tests](https://github.com/metabase/metabase/pull/30356/commits/ab3b821e6ce9befb24e94a76b12ed4f166e07921)